pkg/apis/hubcluster/v1alpha1/types.go

```diff
@@ -0,0 +1,299 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RegisteredCluster represents a cluster known to this hub. The name does NOT match the cluster itself.  Instead we
+// recommend using a generated name to avoid conflicts. The name will get a `cluster-` prefix to map to its namespace.
```

Comment:
Does this imply creation of an individual namespace in the hub per spoke cluster that joins?

Reply:
There is also a `JoinedCluster` API that we added to register a spoke cluster with the hub in the last sprint. Is the `RegisteredCluster` API supposed to be additive of the `JoinedCluster` API?

Reply:
> Does this imply creation of an individual namespace in the hub per spoke cluster that joins?

Yes.

Reply:
> There is also a `JoinedCluster` API that we added to register a spoke cluster with the hub in the last sprint. Is the `RegisteredCluster` API supposed to be additive of the `JoinedCluster` API?

I didn't think to check the doc on the naming guidance you have.  In order to use these for namespace names with a prefix, the naming restrictions are pretty strict.